mmeval/metrics/lvis.py

```diff
@@ -0,0 +1,331 @@
+                 logger: Optional[Logger] = None,
+                 **kwargs) -> None:
+        if not HAS_LVISAPI:
+            raise RuntimeError(
+                'Package lvis is not installed. Please run "pip install '
+                'git+https://github.com/lvis-dataset/lvis-api.git".')
+        super().__init__(
+            metric=metric,
+            classwise=classwise,
+            iou_thrs=iou_thrs,
+            metric_items=metric_items,
+            format_only=format_only,
+            outfile_prefix=outfile_prefix,
+            backend_args=backend_args,
+            **kwargs)
+        self.proposal_nums = proposal_nums  # type: ignore
+
+        with get_local_path(
+                filepath=ann_file, backend_args=backend_args) as local_path:
+            self._lvis_api = LVIS(local_path)
+
+        self.logger = logging.getLogger(__name__) if logger is None else logger
+
+    def add_predictions(self, predictions: Sequence[Dict]) -> None:
+        """Add predictions only.
+
+        If the `ann_file` has been passed, we can add predictions only.
+
+        Args:
+            predictions (Sequence[dict]): Refer to
+                :class:`LVISDetection.add`.
+        """
+        assert self._lvis_api is not None, 'The `ann_file` should be ' \
+            'passesd when use the `LVISDetection.add_predictions` ' \
+            'method, otherwisw use the `LVISDetection.add` instead!'
+        self.add(predictions, groundtruths=[{}] * len(predictions))
+
+    def __call__(self, *args, **kwargs) -> Dict:
+        """Stateless call for a metric compute."""
+
+        # cache states
+        cache_results = self._results
+        cache_lvis_api = self._lvis_api
+        cache_cat_ids = self.cat_ids
+        cache_img_ids = self.img_ids
+
+        self._results = []
+        self.add(*args, **kwargs)
+        metric_result = self.compute_metric(self._results)
+
+        # recover states from cache
+        self._results = cache_results
+        self._lvis_api = cache_lvis_api
+        self.cat_ids = cache_cat_ids
+        self.img_ids = cache_img_ids
+
+        return metric_result
+
+    def compute_metric(self, results: list) -> Dict[str, Union[float, list]]:
+        """Compute the LVIS metrics.
+
+        Args:
+            results (List[tuple]): A list of tuple. Each tuple is the
+                prediction and ground truth of an image. This list has already
+                been synced across all ranks.
+
+        Returns:
+            dict: The computed metric. The keys are the names of
+            the metrics, and the values are corresponding results.
+        """
+        tmp_dir = None
+        if self.outfile_prefix is None:
+            tmp_dir = tempfile.TemporaryDirectory()
+            outfile_prefix = osp.join(tmp_dir.name, 'results')
+        else:
+            outfile_prefix = self.outfile_prefix
+
+        # split gt and prediction list
+        preds, gts = zip(*results)
```

Comment:
`gts` are not used in the following codes.
I think a reasonable workaround should be like, if `gts` are not empty `dict`s, then dump `gts` to an annotation file and create a new `lvis_api` from it to evaluate the result.

Reply:
We do not support load LVIS gt without the annotation file.